plugins/content-manager/src/main/java/com/wazuh/contentmanager/index/ContextIndex.java

```diff
@@ -18,303 +18,118 @@
-     * @throws IOException thrown by XContentFactory.jsonBuilder()
-     */
-    private IndexRequest createIndexRequest(Document document, String id) throws IOException {
-        return new IndexRequest()
-                .index(INDEX_NAME)
-                .source(document.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS))
-                .id(id)
-                .create(true);
-    }
-
-    /**
-     * Get a document from the index.
-     *
-     * @param contextName the id of the document
-     * @return RestStatus of get
-     */
-    public CompletableFuture<RestStatus> get(String contextName) {
-        final CompletableFuture<RestStatus> future = new CompletableFuture<>();
-        final ExecutorService executor = this.threadPool.executor(ThreadPool.Names.WRITE);
-        final TermQueryBuilder termQueryBuilder = QueryBuilders.termQuery("_id", contextName);
-        this.searchSourceBuilder.query(termQueryBuilder);
-
-        executor.submit(
-                () -> {
-                    try (ThreadContext.StoredContext ignored =
-                            this.threadPool.getThreadContext().stashContext()) {
-                        SearchRequest searchRequest =
-                                createSearchRequest(this.searchSourceBuilder.trackTotalHits(true));
-
-                        final SearchResponse searchResponse = this.client.search(searchRequest).actionGet();
-
-                        log.info("Result SEARCH: {}", searchResponse.toString());
-
-                        final RestStatus restStatus = searchResponse.status();
-
-                        future.complete(restStatus);
-
-                        log.info(
-                                "Found {} documents",
-                                Objects.requireNonNull(searchResponse.getHits().getTotalHits()).value);
-                    } catch (Exception e) {
-                        log.error("Error creating SearchRequest due to {}", e.getMessage());
-                        future.completeExceptionally(e);
+                    @Override
+                    public void onFailure(Exception e) {
+                        log.error(
+                                "Failed to index CTI Catalog Context {}, Exception: {}",
+                                consumerInfo.getContext(),
+                                e);
                     }
                 });
-        return future;
     }
 
     /**
-     * Get all documents
+     * Get a context off its index
      *
-     * @return el RestStatus of get all documents
+     * @param contextName ID of the context to be retrieved
+     * @return A completable future holding the response of the query
      */
-    public CompletableFuture<RestStatus> getAll() {
-        final CompletableFuture<RestStatus> future = new CompletableFuture<>();
-        final ExecutorService executor = this.threadPool.executor(ThreadPool.Names.WRITE);
+    public CompletableFuture<SearchResponse> get(String contextName) {
```

Comment:
This method needs to return an exact hit

Consider this JSON
```json
[
  {
    "_index": "wazuh-context",
    "_id": "context-foo",
    "_source": {
      "consumer-bar-A": {
        "offset": 0,
        "last_offset": 44,
        "snapshot": "uri-to-snapshot"
      },
      "consumer-bar-B": {
        "offset": 0,
        "last_offset": 44,
        "snapshot": "uri-to-snapshot"
      }
    }
  }
]
```

Then, we'll want to get one of the consumers. For example, "consumer-bar-A". The method's interface would then be `get(String content, String consumer)`